.github/workflows/cd-pipeline.yaml

```diff
@@ -112,50 +111,91 @@ jobs:
-      - name: Fetch chart and update index
+          # jq
+          sudo curl -SsLf https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64 -o /usr/local/bin/jq && sudo chmod +x /usr/local/bin/jq
+      - id: update-index
+        name: Fetch chart and update index
         run: |
+          # Extract chart release metadata from the publish result file
+          vib_publish_result_file=$(find ~/artifacts -name "result.json" -print -quit)
+          chart_name=$(jq -re '.actions|map(select(.action_id == "helm-publish"))[0] | .application.name' $vib_publish_result_file)
+          chart_version=$(jq -re '.actions|map(select(.action_id == "helm-publish"))[0] | .application.version' $vib_publish_result_file)
+          # Download published asset
           mkdir download
-          chart="${{ needs.get-chart.outputs.chart }}"
-          # HACK: Obtain chart tarball out of the latest commit
-          chart_version="$(yq ".version" charts/bitnami/${chart}/Chart.yaml)"
-          aws s3 cp s3://${{ secrets.AWS_S3_BUCKET }}/bitnami/${chart}-${chart_version}.tgz download/
-          # Rebuild index
-          helm repo index --url https://charts.bitnami.com/bitnami --merge index/bitnami/index.yaml download
-          cp download/index.yaml index/bitnami/index.yaml
+          aws s3 cp s3://${{ secrets.AWS_S3_BUCKET }}/bitnami/${chart_name}-${chart_version}.tgz download/
+
           cd index
-          # Push changes
           git config user.name "Bitnami Containers"
           git config user.email "bitnami-bot@vmware.com"
-          git add bitnami/index.yaml && git commit -m "${chart}-${chart_version}: Update index.yaml" -s && git push
+
+          attempts=0
+          max_attempts=5
+          is_index_updated=0
+          while [[ $attempts -lt $max_attempts && $is_index_updated -eq 0 ]]; do
+            attempts=$((attempts + 1))
+            git fetch origin index
+            git reset --hard origin/index
+
+            # Rebuild index
+            helm repo index --url https://charts.bitnami.com/bitnami --merge bitnami/index.yaml ../download
+            cp ../download/index.yaml bitnami/index.yaml
+            # Push changes
+            git add bitnami/index.yaml && git commit -m "${chart_name}-${chart_version}: Update index.yaml" -s
+            git push && is_index_updated=1 || echo "Failed to push during attempt $attempts"
+          done
+
+          if [[ $is_index_updated -eq 1 ]]; then
+            echo "::set-output name=result::ok"
+          else
+            echo "::set-output name=result::fail"
+          fi
+      - id: show-error
+        name: 'Show error'
+        if: ${{ steps.update-index.outputs.result == 'fail' }}
+        uses: actions/github-script@v3
+        with:
+          script: |
+              core.setFailed('There was a problem updating the index.')
```

Comment:
Is there a need to use the `output.result` here? Given that it can only succeed or fail, I think we can just make the job finish (when it succeeds) or fail (maybe a `exit 1`). That way we avoid both the output and the `show-error` job.

Reply:
As we are not printing specific execution errors here, I think you're right to remove the `show-error` step. Updated.